signer/src/network/mod.rs

```diff
@@ -41,4 +133,70 @@ mod tests {
 
         testing::network::assert_clients_can_exchange_messages(client_1, client_2).await;
     }
+
+    #[tokio::test]
+    async fn two_clients_should_be_able_to_exchange_messages_given_a_libp2p_network() {
+        let _ = tracing_subscriber::fmt()
+            .with_env_filter(tracing_subscriber::EnvFilter::from_default_env())
+            .try_init();
+
+        let key1 = PrivateKey::new(&mut rand::thread_rng());
+        let key2 = PrivateKey::new(&mut rand::thread_rng());
+
+        // TODO: I'm not sure why we need to set the env vars here, when you run this test
+        // individually or with `cargo nextest run` it works fine without, but with
+        // `cargo test -- --test-threads 1` and `make test` it fails with a config
+        // error that the private key is only 2 bytes long .oO
```

Comment:
Weird. Can we create a follow-up issue to track this? Or do you plan to address it before merging in this PR?

Reply:
I'm not sure there's anything to address, I played with this for a while.  Moving to `nextest` is the solution :wink:

Reply:
Fixed it, actually. I changed `SignerContext::init()` to take a `Settings` instead of the path (which I was planning on doing anyway, since we will likely need `Settings` in `main()` to setup logging) -- that seems to have done it :shrug: